chains/evm/calls/events/events.go

```diff
@@ -17,10 +22,69 @@ func (es EventSig) GetTopic() common.Hash {
 const (
 	StartKeygenSig EventSig = "StartKeygen()"
 	KeyRefreshSig  EventSig = "KeyRefresh(string)"
+
+	AcrossDepositSig EventSig = "FundsDeposited(bytes32,bytes32,uint256,uint256,uint256,uint256,uint32,uint32,uint32,bytes32,bytes32,bytes32,bytes)"
 )
 
 // Refresh struct holds key refresh event data
 type Refresh struct {
 	// SHA1 hash of topology file
 	Hash string
 }
+
+type AcrossDeposit struct {
+	InputToken          [32]byte
+	OutputToken         [32]byte
+	InputAmount         *big.Int
+	OutputAmount        *big.Int
+	DestinationChainId  *big.Int
+	DepositId           *big.Int
+	QuoteTimestamp      uint32
+	ExclusivityDeadline uint32
+	FillDeadline        uint32
+	Depositor           [32]byte
+	Recipient           [32]byte
+	ExclusiveRelayer    [32]byte
+	Message             []byte
+}
+
+func (a *AcrossDeposit) ToV3RelayData(originChainID *big.Int) *AcrossV3RelayData {
+	return &AcrossV3RelayData{
+		Depositor:           a.Depositor,
+		Recipient:           a.Recipient,
+		ExclusiveRelayer:    a.ExclusiveRelayer,
+		InputToken:          a.InputToken,
+		OutputToken:         a.OutputToken,
+		InputAmount:         a.InputAmount,
+		OutputAmount:        a.OutputAmount,
+		OriginChainId:       originChainID,
+		DepositId:           a.DepositId,
+		FillDeadline:        a.FillDeadline,
+		ExclusivityDeadline: a.ExclusivityDeadline,
+		Message:             a.Message,
+	}
+}
+
+type AcrossV3RelayData struct {
+	Depositor           [32]byte
+	Recipient           [32]byte
+	ExclusiveRelayer    [32]byte
+	InputToken          [32]byte
+	OutputToken         [32]byte
+	InputAmount         *big.Int
+	OutputAmount        *big.Int
+	OriginChainId       *big.Int
+	DepositId           *big.Int
+	FillDeadline        uint32
+	ExclusivityDeadline uint32
+	Message             []byte
+}
+
+func (a *AcrossV3RelayData) Calldata(repaymentChainID *big.Int, repaymentAddress common.Address) ([]byte, error) {
+	abi, _ := abi.JSON(strings.NewReader(consts.SpokePoolABI))
```

Comment:
why not move this into global variable? Loading and parsing abi does require some compute units